astropy/coordinates/transformations.py

```diff
@@ -741,8 +744,104 @@ def __call__(self, fromcoord, toframe):
                 'should have been of type {1}'.format(res, self.tosys))
         return res
 
+class BaseAffineTransform(CoordinateTransform):
+    """
+    Base class for common functionality between the ``AffineTransform``-type
+    subclasses.
+    """
+
+    def _apply_transform(self, fromcoord, matrix, vectors):
+        from .representation import (CartesianRepresentation,
+                                     UnitSphericalRepresentation,
+                                     CartesianDifferential)
+
+        rep = fromcoord.represent_as(CartesianRepresentation)
+        newrep = rep.without_differentials()
+
+        # Only do transform is matrix is specified. This is for speed in
+        # transformations that only specify an offset (e.g., LSR)
+        if matrix is not None:
+            newrep = newrep.transform(matrix)
+
+        if vectors is not None and vectors[0] is not None:
+            pos_offset = CartesianRepresentation(vectors[0])
```

Comment:
Would it not be reasonable to assume the vectors are `CartesianRepresentation` already? For `get_body_barycentric`, this is in fact the output one gets, so there now are two transform to a quantity and back here to a representation that are not really needed.